### PR TITLE
PlanckianJitter implementation

### LINKED ISSUE #1244: Add Planckian jitter augmentation
Proposed in paper _"Planckian jitter: enhancing the color quality of self-supervised visual representations"_
Link: https://arxiv.org/pdf/2202.07993.pdf
[PyTorch code as a reference](https://kornia.readthedocs.io/en/latest/_modules/kornia/augmentation/_2d/intensity/planckian_jitter.html#RandomPlanckianJitter)
Code from authors: https://github.com/TheZino/PlanckianJitter

Example of ColorJitter vs PlanckianJitter from authors.
image

### LINKED ISSUE #1540: [Feature request] Add Transforms from Kornia
Kornia library has an extensive list of transforms that:
* May not exist in Albumentations
* Exist but have better implementation

We need to check out that we are on top of the situation:
---
https://kornia.readthedocs.io/en/latest/augmentation.html
---

- [ ] GaussianIllumination
- [x] PlanckianJitter https://arxiv.org/pdf/2202.07993.pdf
- [ ] PlasmaBrightnessContrast https://arxiv.org/abs/2204.03776
- [ ] RandomPlasmaShadow https://arxiv.org/abs/2204.03776

### PR DESCRIPTION
**Pull request by zakajd**
Closes #1244 and partially closes #1540

## Summary by Sourcery

This pull request introduces the PlanckianJitter transformation for image augmentation, which simulates realistic variations in chromaticity by jittering the image illuminant along the Planckian locus. It includes a new function planckian_jitter and comprehensive documentation for the new transformation.

* **New Features**:
    - Introduced the PlanckianJitter transformation for image augmentation, which simulates realistic variations in chromaticity by jittering the image illuminant along the Planckian locus.
* **Enhancements**:
    - Added a new function planckian_jitter to perform the Planckian Jitter transformation on images, supporting both 'blackbody' and 'CIED' modes with temperature ranges.
* **Documentation**:
    - Added documentation for the PlanckianJitter transformation, including its parameters, target image types, and references.

### CODE REVIEW

**Comment by zakajd** [comment]
Hey, @ternaus @Dipet let's review this one quickly. 

The implementation is slightly different from original one in referenced repo. 
1. Changer sampling of temperature values to allow temperature interpolation
2. Also original code samples cold temperatures (> 6000K) way to often, fixed that with a two-case sampling process.

**Comment by ternaus** [comment]
@zakajd It look like you are online :)

Could you please add InitSchema class from the https://github.com/albumentations-team/albumentations/pull/1790/files#diff-43763383dfbb0d910738679d562a14ae3707a89dde83530ba6115a2547cb6772R3495

Add to this PR, I will merge this one and close that one.

File: albumentations/augmentations/transforms.py [hunk 1]
```diff
@@ -3453,3 +3453,86 @@ def targets(self) -> Dict[str, Callable[..., Any]]:
...
+
+    Reference:
+        - https://github.com/TheZino/PlanckianJitter
+        - https://arxiv.org/pdf/2202.07993.pdf
+
+    """
+
+    def __init__(
+        self,
+        mode: str = "blackbody",
```
Comment (ternaus) [inline albumentations/augmentations/transforms.py:3484]: `mode: str = "blackbody"` => `mode: Literal["CIED", "blackbody"] = "blackbody"`

File: albumentations/augmentations/transforms.py [hunk 2]
```diff
@@ -3453,3 +3453,86 @@ def targets(self) -> Dict[str, Callable[..., Any]]:
...
+        temperature_limit: Tuple[int, int] = (3000, 15000),
+        sampling_method: str = "uniform",
```
Comment (ternaus) [inline albumentations/augmentations/transforms.py:3486]: =>
`sampling_method: Literal["uniform", "gaussian"]`

File: albumentations/augmentations/functional.py [hunk 3]
```diff
@@ -1432,3 +1432,81 @@ def center(width: NumericType, height: NumericType) -> Tuple[float, float]:
...
+        13_000: [0.3421, 0.4719, 0.6631],
+        13_500: [0.3365, 0.4719, 0.6702],
+        14_000: [0.3313, 0.4719, 0.6766],
+        14_500: [0.3263, 0.4719, 0.6826],
+        15_000: [0.3217, 0.4719, 0.6882],
+    },
+}
+
+
+def planckian_jitter(img: np.ndarray, temperature: int, mode: str = "blackbody") -> np.ndarray:
```
Comment (ternaus) [inline albumentations/augmentations/functional.py:1493]: `mode: Literal["blackbody", "CIED"]`

**Review by ternaus — CHANGES_REQUESTED** [review]

**Comment by ternaus** [comment]
Addressed in https://github.com/albumentations-team/albumentations/pull/1790

**Pull request closed without merging**